### PR TITLE
Added list gist command.

### LINKED ISSUE #5: Add command to list all public gist
This command should fetch all public gists of a user.

Suggested command:

gac list gist <username>

Output: Could be a list or table of content.

Reference:

[API for fetching repo](https://api.github.com/users/thamaraiselvam/gists)

### PR DESCRIPTION
**Pull request by samit22**
Command: list gist
Should close https://github.com/thamaraiselvam/github-api-cli/issues/5

### CODE REVIEW

File: cmd/publicGist.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package cmd
+
+import (
```
Comment (thamaraiselvam) [inline cmd/publicGist.go]: Move this command under list command, check followers command for reference

File: cmd/root.go [hunk 2]
```diff
@@ -2,15 +2,16 @@ package cmd
...
+
+	"github.com/spf13/cobra"
 )
 
 var rootCmd = &cobra.Command{}
 
 //Execute on rootCmd
 func Execute() {
-	commands := []func() *cobra.Command{infoCmd}
+	commands := []func() *cobra.Command{infoCmd, publicGistsCmd}
```
Comment (thamaraiselvam) [inline cmd/root.go]: This change is not necessary once you moved gist command under list

File: cmd/service/service.go [hunk 3]
```diff
@@ -60,6 +76,9 @@ func makeRequest(method string, URL string, body io.Reader) (*http.Response, err
 	if resp.StatusCode == 404 {
 		return nil, fmt.Errorf("user not found")
 	}
+	if resp.StatusCode == 500 {
```
Comment (thamaraiselvam) [inline cmd/service/service.go]: In the latest changes, If the status code is not 200. we just return status whatever we get.

File: cmd/publicGist.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		Long:  "Retrieve public gists with owner github id, description, created date and url",
+		Args:  nil,
+		Run: func(cmd *cobra.Command, args []string) {
+			getPublicGistList()
+		},
+	}
+}
+
+func getPublicGistList() {
+	client := service.CreateClient("/gists/public")
```
Comment (thamaraiselvam) [inline cmd/publicGist.go]: How you are forming URL with given **<username>** as an argument?

File: cmd/types/types.go [hunk 5]
```diff
@@ -6,3 +6,11 @@ type UserInfo struct {
 	Location    string `json:"location"`
 	PublicRepos int    `json:"public_repos"`
 }
+
+//PublicGist contains basic info of publi gist
+type PublicGist struct {
```
Comment (thamaraiselvam) [inline cmd/types/types.go]: Currently, the entire app is fetching only public data so please rename it into Gist

Comment (thamaraiselvam) [reply cmd/types/types.go]: And you might need gists list to store gist

**Review by thamaraiselvam — COMMENTED** [review]
I really appreciate your PR. 

I have added a few comments, Please fix

**Review by thamaraiselvam — CHANGES_REQUESTED** [review]
Please fix comments

**Comment by thamaraiselvam** [comment]
Can you please resolve conflicts

**Comment by thamaraiselvam** [comment]
Build is failing.

File: cmd/service/service.go [hunk 6]
```diff
@@ -3,18 +3,23 @@ package service
...
+
+	"github.com/thamaraiselvam/git-api-cli/cmd/types"
 )
 
 const githubURL = "https://api.github.com"
 
 //Client is interface of service
 type Client interface {
 	GetUser() (types.UserInfo, error)
+<<<<<<< HEAD
```
Comment (thamaraiselvam) [inline cmd/service/service.go]: pushed along with conflicts?

File: cmd/types/types.go [hunk 7]
```diff
@@ -7,6 +7,14 @@ type UserInfo struct {
 	PublicRepos int    `json:"public_repos"`
 }
 
+//Gist contains basic info of publi gist
+type Gist struct {
```
Comment (thamaraiselvam) [inline cmd/types/types.go]: this struct should be singular and can be named as `gist` then have list of `gist` called `Gists`

File: cmd/info.go [hunk 8]
```diff
@@ -0,0 +1,40 @@
+package cmd
+
+import (
+	"fmt"
```
Comment (thamaraiselvam) [inline cmd/info.go]: this file needs to be removed since it is already moved inside of info dir

File: cmd/root.go [hunk 9]
```diff
@@ -2,10 +2,13 @@ package cmd
 
 import (
 	"fmt"
+	"os"
+
 	"github.com/spf13/cobra"
 	"github.com/thamaraiselvam/git-api-cli/cmd/info"
 	"github.com/thamaraiselvam/git-api-cli/cmd/list"
-	"os"
+
```
Comment (thamaraiselvam) [inline cmd/root.go]: This change is not necessary since `github.com/spf13/cobra` is already added

**Comment by samit22** [comment]
@thamaraiselvam This is work in progress.

**Review by thamaraiselvam — APPROVED** [review]

**Comment by thamaraiselvam** [comment]
Thanks for the PR @samit22  👏🏼

**Pull request merged**